src/types/box-types.ts

```diff
@@ -3,24 +3,35 @@ import { EnhancerProps } from './enhancers'
 
 /**
  * @template T Object
- * @template K Union of T keys
+ * @template K Union of keys (not necessarily present in T)
  */
-export type Without<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+export type Without<T, K> = Pick<T, Exclude<keyof T, K>>
 
 /**
  * "is" prop
  * @template P Props
  */
 export type Is<P = any> = React.ElementType<P>
 
+/**
+ * Remove box props from object `T` if they're present
+ * @template T Object
+ */
+export type WithoutBoxProps<T> = Without<T, "is" | "innerRef">
```

Comment:
I think this name is a bit misleading because the type will still have all the Box CSS props, including `clearfix` which isn't even a valid CSS prop 😅

Reply:
`WithoutBoxProps` doesn't produce a type with all those Box CSS props or `clearfix` though 🤔 unless you pass that to it (which we aren't doing).

Reply:
@Rowno do you have a suggestion here?

Reply:
The name to me implies that you can pass it a component that extends Box and it'll remove all the Box props, leaving only the component's own props. Since we're exporting the type, people might try to use it.

Reply:
Removed the export here.